test_validation.py

```diff
@@ -8,15 +8,43 @@ def setUp(self):
         self.test_board = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
         self.validator = Validator
 
-    def test_validate_selection(self):
-        with self.subTest('returns true when in range integers'):
-            result = self.validator.validate_selection(1, self.test_board)
+    def test_is_on_board(self):
+        test_board = [[1, 2, 3], ['x', 5, 6], [7, 8, 9]]
+        with self.subTest('returns true when selection is available'):
+            result = self.validator.is_on_board(1, test_board)
             self.assertTrue(result)
 
-        with self.subTest('returns false when out of range integers'):
-            result = self.validator.validate_selection(10, self.test_board)
+        with self.subTest('returns false when selection is not available'):
+            result = self.validator.is_on_board(4, test_board)
             self.assertFalse(result)
 
-        with self.subTest('returns false when non integers'):
-            result = self.validator.validate_selection("r", self.test_board)
+    def test_is_in_range(self):
+        board_range = range(1, len(self.test_board) * len(self.test_board) + 1)
```

Comment:
really nice job using logical describe blocks, consistent patterns in your names, and descriptive test cases/outcomes 👏👏👏

Reply:
+1000

Reply:
I also really love that you took the feedback for PRs and ran with it, consistently with every PR you have done since. The Descriptions are great.